### ISSUE TITLE
Form validation fails if javascript is disabled

### ISSUE THREAD

**Issue by rubenquidam** [bcit-ci/CodeIgniter#5683]
When using custom values for the pattern attribute (e.g. "alpha_numeric" instead of a regex), form validation would fail if JavaScript is disabled. Eg: https://amazonwatch.org/take-action/urge-ecuadors-new-oil-friendly-environmental-minister-to-reject-yasuni-oil-drilling (not my site).

**Comment by rubenquidam** [comment]
Browser side FV doesn't use JS, the problem is that CI uses a value that is not in the spec (alpha_numeric) and it gets dynamically built into a regex using JS. As a result, forms validated with values such as alpha_numeric only work if JS is enabled.

**Comment by narfbg** [comment]
My previous comment was not an exaggeration - `CI_Form_validation` does not contain a single line, or even a tiny chunk of JavaScript.

I have no idea where you're getting these ideas/conclusions from, but you're undoubtedly incorrect about them. What you're saying is simply impossible.

**Comment by rubenquidam** [comment]
You are right, my apologies. I found the culprit to be foundation.js, which uses the same pattern name for form validation ("alpha_numeric" among others) as codeigniter, which pointed me in the wrong direction! Sorry about that.
